### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by smola**

### CODE REVIEW

**Comment by smola** [comment]
This PR fixes https://github.com/src-d/go-git/issues/246 and partially https://github.com/src-d/go-git/issues/242

File: plumbing/object/tag.go [hunk 1]
```diff
@@ -20,9 +20,13 @@ import (
...
-	Tagger     Signature
+	// Hash of the tag.
+	Hash plumbing.Hash
+	// Name of the tag.
+	Name string
+	// Tagger is the one who created the tag.
+	Tagger Signature
+	// Message is an arbitrary text message
 	Message    string
 	TargetType plumbing.ObjectType
```
Comment (ajnavarro) [inline plumbing/object/tag.go]: Can you add comments to this two properties too?

File: plumbing/object/tag.go [hunk 2]
```diff
@@ -20,9 +20,13 @@ import (
 //
```
Comment (ajnavarro) [inline plumbing/object/tag.go]: Add some info about [Lightweight Tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging#_lightweight_tags) and say that this method does not return this kind, can help people that don't know this two kinds of tags.

Comment (smola) [inline plumbing/object/tag.go]: Done.

File: plumbing/object/tag.go (hunk 1 above)
Comment (smola) [inline plumbing/object/tag.go]: Done.

**Pull request merged**